js/apps/admin-ui/src/authentication/components/modals/AddStepModal.tsx

```diff
@@ -142,7 +143,7 @@ export const AddStepModal = ({ name, type, onSelect }: AddStepModalProps) => {
           }}
           inputGroupName="search"
           inputGroupPlaceholder={t("search")}
-          inputGroupOnEnter={setSearch}
+          inputGroupOnEnter={(_) => setSearch(_.toLowerCase())}
```

Comment:
Why set this to lowercase if the search itself is already being done in a case-insensitive manner?

Reply:
if I don't set the input as lowercase the search wouldn't work for uppercase or alternating caps inputs. Below you can see an example removing the lowercase (sorry for the quality of the edit).

Let me know if I'm missing something.

![search](https://github.com/keycloak/keycloak/assets/26752730/f7d9ab73-bf81-4fd1-9c1c-d3a2a99ed7fc)

Reply:
Ah my mistake. What I meant is rather than calling `.toLowerCase()` when setting the value, it should be done when searching. So in the other logic you changed you can call `search.toLowerCase()`.

This prevents accidental regressions if the value is ever set somewhere else, where one might forget to call `.toLowerCase()`.

Reply:
Moving the .toLowerCase() to the TableToolbar component (which is the one who does the search) involved modify the other componentes that use it. Also an exactSearch attribute was added for group searches because those are not case-insensitive in the backend.